### ISSUE TITLE
Agent does not perform DNS lookup for manager when reconnecting to manager

### ISSUE THREAD

**Issue by mrtc0** [wazuh/wazuh#4817]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.0 | ossec-agentd | Agent | Packages | Ubuntu 18.04 |

The wazuh-agent (ossec-agent) does not do DNS resolution of wazuh-manager address when reconnecting to wazuh-manager.
This mean that if wazuh-manager goes down due to an incident or something, and then manager's IP address changed, wazuh-agent has to be restarted to connect to wazuh-manager.

## Step to reproduce

Use the `/etc/hosts` to reproduce the wazuh-manager's IP address being disconnected.
First, write the address of the wazuh-manager that you have been connected to so far.

```
192.168.1.1 wazuh-manager.mydomain.local
```

Restart the wazuh-agent.
The wazuh-manager has already been changed to a different IP address, so naturally, agent can't connect.

```
2020/04/01 12:31:39 ossec-agentd: ERROR: (1216): Unable to connect to '192.168.1.1': 'Connection timed out'.
2020/04/01 12:31:49 ossec-agentd: INFO: Trying to connect to server (
wazuh-manager.mydomain.local/192.168.1.1:1514/tcp).
```

Then write the correct wazuh-manager address to `/etc/hosts` .

```
10.0.1.1 wazuh-manager.mydomain.local
```

wazuh-agent will continue to try to connect to your previous IP address unless restart it.

```
2020/04/01 12:56:49 ossec-agentd: ERROR: (1216): Unable to connect to '192.168.1.1': 'Connection timed out'.
2020/04/01 12:56:49 ossec-agentd: INFO: Trying to connect to server (
wazuh-manager.mydomain.local/192.168.1.1:1514/tcp).
```

**Comment by Lopuiz** [comment]
Hello @mrtc0 

It fixed in 3.11.4 version. In 3.12 version should works (I have tested and it works). Please, could you confirm that the agent is in 3.12 version?

Regards,
Eva

**Comment by mrtc0** [comment]
Thank you for your reply.
It was my mistake, and when I checked the version again, it was 3.11.0.
When I updated to 3.12.0 and checked, it worked. Thank you.
